### ISSUE TITLE
Python3.8-APScheduler (resp. vorta) fails with "trollius-2.2.post1 not supported for interpreter python3.8"

### ISSUE THREAD

**Issue by Melkor333** [NixOS/nixpkgs#116400]
**Describe the bug**

TLDR: Apparently the `Python3.8-APScheduler` has `trollius` as a dependency which doesn't work with python3.

I wanted to install the backup tool `vorta` (respectively update it) and got the following error. 
I'm not sure wether the main issue is that `trollius` fails on python 3 or that APScheduler depends on `trollius` as it seems like `trollius` is deprecated and should be replaced.

The stacktrace happens when installing either one of the following packages:
```
nixos.vorta
nixos.python3Packages.trollius
nixos.python36Packages.trollius
nixos.python37Packages.trollius
nixos.python38Packages.trollius
nixos.python39Packages.trollius
nixos.python38Packages.APScheduler


✦ at 11:25:17 ❯ nix-env -iA nixos.python38Packages.trollius --show-trace 
error: while evaluating 'callPackageWith' at /nix/store/6dx109bn6b0s9drkrzjnlbs9ad0acq79-nixos-21.05pre275822.916ee862e87/nixos/lib/customisation.nix:117:35, called from /nix/store/6dx109bn6b0s9drkrzjnlbs9ad0acq79-nixos-21.05pre275822.916ee862e87/nixos/pkgs/top-level/python-packages.nix:8316:14:
while evaluating 'makeOverridable' at /nix/store/6dx109bn6b0s9drkrzjnlbs9ad0acq79-nixos-21.05pre275822.916ee862e87/nixos/lib/customisation.nix:67:24, called from /nix/store/6dx109bn6b0s9drkrzjnlbs9ad0acq79-nixos-21.05pre275822.916ee862e87/nixos/lib/customisation.nix:121:8:
while evaluating anonymous function at /nix/store/6dx109bn6b0s9drkrzjnlbs9ad0acq79-nixos-21.05pre275822.916ee862e87/nixos/pkgs/development/python-modules/trollius/default.nix:1:1, called from /nix/store/6dx109bn6b0s9drkrzjnlbs9ad0acq79-nixos-21.05pre275822.916ee862e87/nixos/lib/customisation.nix:69:16:
while evaluating 'makeOverridablePythonPackage' at /nix/store/6dx109bn6b0s9drkrzjnlbs9ad0acq79-nixos-21.05pre275822.916ee862e87/nixos/pkgs/top-level/python-packages.nix:29:37, called from /nix/store/6dx109bn6b0s9drkrzjnlbs9ad0acq79-nixos-21.05pre275822.916ee862e87/nixos/pkgs/development/python-modules/trollius/default.nix:3:1:
while evaluating 'makeOverridable' at /nix/store/6dx109bn6b0s9drkrzjnlbs9ad0acq79-nixos-21.05pre275822.916ee862e87/nixos/lib/customisation.nix:67:24, called from /nix/store/6dx109bn6b0s9drkrzjnlbs9ad0acq79-nixos-21.05pre275822.916ee862e87/nixos/pkgs/top-level/python-packages.nix:31:12:
while evaluating anonymous function at /nix/store/6dx109bn6b0s9drkrzjnlbs9ad0acq79-nixos-21.05pre275822.916ee862e87/nixos/pkgs/development/interpreters/python/mk-python-derivation.nix:30:1, called from /nix/store/6dx109bn6b0s9drkrzjnlbs9ad0acq79-nixos-21.05pre275822.916ee862e87/nixos/lib/customisation.nix:69:16:
trollius-2.2.post1 not supported for interpreter python3.8
```

The stracktrace first appeared to me when I wanted to `nixos-rebuild switch --upgrade` my system.

**To Reproduce**
1. Check that on up to date `nixos-unstable` branch
2. run `nix-env -iA nixos.python3Packages.trollius`

I assume any `python 3.8` program which depends on APScheduler will fail.

**Expected behavior**
vorta / apscheduler should be installed :)

**Notify maintainers**

CC @SuperSandro2000 you were the last person to touch APScheduler
CC @Ma27 you maintain vorta and might be interested in this aswell

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.100, NixOS, 21.05pre274338.102eb68ceec (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.10`
 - channels(root): `"nixos-21.05pre275822.916ee862e87, nixos-hardware"`
 - channels(sam): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
vorta
python3Packages.trollius
python3Packages.APScheduler
# a list of nixos modules affected by the problem
module:
```

**Comment by marsam** [comment]
addressed by https://github.com/NixOS/nixpkgs/commit/c75c92f536ed32a83081a4725fd77b3162a07c9f
